src/pyats/contrib/creators/tests/test_yamltemplate.py

```diff
@@ -0,0 +1,139 @@
+from unittest import TestCase, main, mock
+from pyats.contrib.creators.yamltemplate import Yamltemplate
+from pyats.topology import Testbed
+
+
+class TestYamltemplate(TestCase):
+
+    def setUp(self):
+        self.tmpl_str = """devices:
+  $device_name:
+    connections:
+      cli:
+        ip: $mgmt_ip
+        protocol: ssh
+    credentials:
+      default:
+        username: $username
+        password: $password
```

Comment:
The pattern we use is slightly different than the one you used here. Can you please change it (and where it is applicable) to match the example below?

```suggestion
        ip: %{mgmt_ip}
        protocol: ssh
    credentials:
      default:
        username: %{username}
        password: %{password}
```

Reply:
Sure, will update.  string.Template allows both $ and ${} (or % and %{} with the new delimiter), so I tested both here.